jubilant/_juju.py

```diff
@@ -393,12 +397,17 @@ def exec(
         else:
             assert unit is not None
             args.extend(['--unit', unit])
+        if wait is not None:
+            args.extend(['--wait', f'{wait}s'])
         args.append('--')
         args.extend(command)
 
         try:
             stdout, stderr = self._cli(*args)
         except CLIError as exc:
+            if 'timed out' in exc.stderr:
+                msg = f'timed out waiting for command, stderr:\n{exc.stderr}'
+                raise TimeoutError(msg) from None
```

Comment:
How about `from exc` to include the `CLIError` in the traceback?
```suggestion
                raise TimeoutError(msg) from exc
```

Reply:
I had that at first, but it just makes the traceback really long without adding any value. All the info you need is there. Compare the current version:

```
>>> juju.run('testdb/0', 'do-thing', wait=0.1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ben/w/jubilant/jubilant/_juju.py", line 558, in run
    raise TimeoutError(msg) from None
TimeoutError: timed out waiting for action, stderr:
Running operation 40 with 1 task
  - task 41 on unit-testdb-0

Waiting for task 41...
ERROR timed out waiting for results from: unit testdb/0
```

With the `raise ... from exc` version:

```
>>> juju.run('testdb/0', 'do-thing', wait=0.1)
Traceback (most recent call last):
  File "/home/ben/w/jubilant/jubilant/_juju.py", line 554, in run
    stdout, stderr = self._cli(*args)
                     ^^^^^^^^^^^^^^^^
  File "/home/ben/w/jubilant/jubilant/_juju.py", line 185, in _cli
    raise CLIError(e.returncode, e.cmd, e.stdout, e.stderr) from None
jubilant._juju.CLIError: Command '['juju', 'run', '--format', 'json', 'testdb/0', 'do-thing', '--wait', '0.1s']' returned non-zero exit status 1.
Stdout:
{}

Stderr:
Running operation 42 with 1 task
  - task 43 on unit-testdb-0

Waiting for task 43...
ERROR timed out waiting for results from: unit testdb/0


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ben/w/jubilant/jubilant/_juju.py", line 558, in run
    raise TimeoutError(msg) from exc
TimeoutError: timed out waiting for action, stderr:
Running operation 42 with 1 task
  - task 43 on unit-testdb-0

Waiting for task 43...
ERROR timed out waiting for results from: unit testdb/0
```

Reply:
Yeah that is extremely verbose isn't it, and I think the only extra piece of information is the exact Juju command that was executed `['juju', 'run', '--format', 'json', 'testdb/0', 'do-thing', '--wait', '0.1s']`